src/io/reader.cpp

```diff
@@ -11,71 +11,109 @@ namespace copc
+}
+
+std::map<uint64_t, las::EvlrHeader> Reader::ReadEvlrs()
+{
+    std::map<uint64_t, las::EvlrHeader> out;
+
+    // Move stream to beginning of VLRs
+    in_stream_->seekg(reader_->header().evlr_offset);
+
+    // Iterate through all evlr's and add them to the `evlrs` list
+    size_t count = 0;
+
+    while (count < reader_->header().evlr_count && in_stream_->good() && !in_stream_->eof())
     {
-        uint64_t cur_pos = this->in_stream_->tellg();
-        las::VlrHeader h = las::VlrHeader::create(*this->in_stream_);
+        uint64_t cur_pos = in_stream_->tellg();
+        las::EvlrHeader h = las::EvlrHeader::create(*in_stream_);
         out[cur_pos] = h;
 
-        this->in_stream_->seekg(h.data_length, std::ios::cur); // jump foward
+        in_stream_->seekg(h.data_length, std::ios::cur); // jump foward
         count++;
     }
     return out;
 }
 
 las::CopcInfoVlr Reader::ReadCopcInfo()
 {
-    this->in_stream_->seekg(COPC_OFFSET);
-    las::CopcInfoVlr copc_info = las::CopcInfoVlr::create(*this->in_stream_);
+    in_stream_->seekg(COPC_OFFSET);
+    las::CopcInfoVlr copc_info = las::CopcInfoVlr::create(*in_stream_);
     return copc_info;
 }
 
-CopcExtents Reader::ReadCopcExtents(const las::CopcInfoVlr &copc_data, const las::LasHeader &header,
-                                    const las::EbVlr &eb_vlr)
+CopcExtents Reader::ReadCopcExtents(const std::map<uint64_t, las::VlrHeader> &vlrs, const las::EbVlr &eb_vlr)
 {
+
     // TODO[Leo]: (Extents) Update this once we have updated copc test file
-    if (header.point_format_id < 6 || header.point_format_id > 8)
+    if (reader_->header().point_format_id < 6 || reader_->header().point_format_id > 8)
         return {7, static_cast<uint16_t>(eb_vlr.items.size())};
 
-    this->in_stream_->seekg(copc_data.extent_vlr_offset);
-
-    return {las::CopcExtentsVlr::create(*this->in_stream_, copc_data.extent_vlr_size), header.point_format_id,
-            static_cast<uint16_t>(eb_vlr.items.size())};
+    for (auto &[offset, vlr_header] : vlrs)
+    {
+        if (vlr_header.user_id == "copc" && vlr_header.record_id == 10000)
+        {
+            in_stream_->seekg(offset + las::VlrHeader::Size);
+            return {las::CopcExtentsVlr::create(*in_stream_, vlr_header.data_length),
+                    static_cast<int8_t>(reader_->header().point_format_id), static_cast<uint16_t>(eb_vlr.items.size())};
+        }
+    }
+    return {static_cast<int8_t>(reader_->header().point_format_id), static_cast<uint16_t>(eb_vlr.items.size())};
 }
 
-las::WktVlr Reader::ReadWktData(const las::CopcInfoVlr &copc_data)
+las::WktVlr Reader::ReadWktData(const std::map<uint64_t, las::VlrHeader> &vlrs)
 {
-    this->in_stream_->seekg(copc_data.wkt_vlr_offset);
-    las::WktVlr wkt = las::WktVlr::create(*this->in_stream_, copc_data.wkt_vlr_size);
-    return wkt;
+    for (auto &[offset, vlr_header] : vlrs)
+    {
+        if (vlr_header.user_id == "LASF_Projection" && vlr_header.record_id == 2112)
```

Comment:
should we make this a helper function? it can return las::VlrHeader (which is agonistic to vlr vs evlr):

```
las::VlrHeader *FetchVlr(user_id, record_id)
{
for (vlr in vlrs)
{
if (vlr matches user_id and record_id)
return
}
return nullptr
}
```

(not sure why github doesn't let you use tabs...)